File: obfuscator/obfuscationFuns/reverseBits.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function reverseBits(input, keyArray, initor) {
    const reversed = new Uint8Array(input.length);
    for (let i = 0; i < input.length; i++) {
        let byte = input[i];
        let reversedByte = 0;
        for (let j = 0; j < 8; j++) {
            reversedByte <<= 1;
            reversedByte |= byte & 1;
            byte >>= 1;
        }
        reversed[i] = reversedByte;
    }
    return reversed;
}
function de_reverseBits(input, keyArray, initor) {
    const length = input.length;
    const deobfuscated = new Uint8Array(length);
    for (let i = 0; i < length; i++) {
        let value = input[i];
        let result = 0;
        for (let j = 0; j < 8; j++) {
            result = (result << 1) | (value & 1);
            value >>= 1;
        }
        deobfuscated[i] = result;
    }
    return deobfuscated;
}
let funPair = {
    obfuscation: reverseBits,
    deobfuscation: de_reverseBits
};
exports.default = funPair;

File: android/wireguard_obfuscation/obfuscationFuns/ReverseBits.java
package com.morph_vpn.plugin.wireguard_obfuscation.obfuscationFuns;
import java.util.Arrays;

public class ReverseBits {
    public static byte[] reverseBits(byte[] input, byte[] keyArray, Object initor) {
        byte[] reversed = new byte[input.length];

        for (int i = 0; i < input.length; i++) {
            byte value = input[i];
            byte reversedByte = 0;

            for (int j = 0; j < 8; j++) {
                reversedByte <<= 1;
                reversedByte |= value & 1;
                value >>= 1;
            }

            reversed[i] = reversedByte;
        }

        return reversed;
    }

    public static byte[] de_reverseBits(byte[] input, byte[] keyArray, Object initor) {
        byte[] deobfuscated = new byte[input.length];

        for (int i = 0; i < input.length; i++) {
            byte value = input[i];
            byte result = 0;

            for (int j = 0; j < 8; j++) {
                result = (byte) ((result << 1) | (value & 1));
                value >>= 1;
            }

            deobfuscated[i] = result;
        }

        return deobfuscated;
    }

    public static void main(String[] args) {
        byte[] input = new byte[]{/* initialize input byte array */};
        byte[] keyArray = new byte[]{/* initialize keyArray byte array */};
        Object initor = null;

        byte[] reversed = reverseBits(input, keyArray, initor);
        byte[] deobfuscated = de_reverseBits(reversed, keyArray, initor);

        System.out.println("Input: " + Arrays.toString(input));
        System.out.println("Reversed: " + Arrays.toString(reversed));
        System.out.println("Deobfuscated: " + Arrays.toString(deobfuscated));
    }
}
